### PR TITLE
docs: Added details on plugin name and plugin alias standards in guide

### PR DESCRIPTION
**Pull request by Aerex**
# Context

The purpose of this PR is to provide clearly expectations to use a proper plugin name and alias when designing a plugin. It is important that plugin's names clearly describe what service the plugin is providing as well as short alias name for uninstall and installing the plugin.

### CODE REVIEW

File: docs/plugin_developer_guide.md [hunk 1]
```diff
@@ -74,6 +74,9 @@ IBM Cloud CLI SDK provides a set of APIs to register and manage plug-ins. It als
 
     **Understanding the fields in this `plugin.PluginMetadata` struct:**
     - _Name_: The name of plug-in. It will be displayed when using `ibmcloud plugin list` command or can be used to uninstall the plug-in through `ibmcloud plugin uninstall` command.
+      - It **strongly** encouraged to use a name that best describes the service the plugin is provided.
```
Comment (steveclay) [inline docs/plugin_developer_guide.md:77]: It **is** strongly ...

Also, when we describe plug-ins, we write them with the hyphen. Only the `plugin` command lacks the hyphen.

File: docs/plugin_developer_guide.md [hunk 2]
```diff
@@ -74,6 +74,9 @@ IBM Cloud CLI SDK provides a set of APIs to register and manage plug-ins. It als
...
+    - _Aliases_: A list of short names of the plugin's name that can be used as a stand-in for installing or uninstall the plugin.
```
Comment (steveclay) [inline docs/plugin_developer_guide.md:78]: ... A list of short names of the plug-in that can be used as a stand-in for installing, updating, uninstalling and using the plug-in.

File: docs/plugin_developer_guide.md [hunk 3]
```diff
@@ -74,6 +74,9 @@ IBM Cloud CLI SDK provides a set of APIs to register and manage plug-ins. It als
...
+      - It is strongly recommended that you have at least one alias to better usability of the plugin.
```
Comment (steveclay) [inline docs/plugin_developer_guide.md:79]: "to better" replaced with "to improve the". And change plugin to plug-in

File: docs/plugin_developer_guide.md (hunk 1 above)
Comment (Aerex) [reply docs/plugin_developer_guide.md:77]: Cool. I will keep that in mind.

File: docs/plugin_developer_guide.md [hunk 4]
```diff
@@ -74,6 +74,9 @@ IBM Cloud CLI SDK provides a set of APIs to register and manage plug-ins. It als
 
     **Understanding the fields in this `plugin.PluginMetadata` struct:**
     - _Name_: The name of plug-in. It will be displayed when using `ibmcloud plugin list` command or can be used to uninstall the plug-in through `ibmcloud plugin uninstall` command.
+      - It is **strongly** encouraged to use a name that best describes the service the plug-in is provided.
```
Comment (steveclay) [inline docs/plugin_developer_guide.md:77]: One grammar tweak:  `... is provided` change to `provides` Otherwise looks good 👍 .

**Review by steveclay — APPROVED** [review]

**Pull request merged**